code/implementation/system-stack/src/main/java/io/cattle/platform/systemstack/catalog/impl/CatalogServiceImpl.java

```diff
@@ -247,13 +249,15 @@ protected void refresh() throws IOException {
         if (firstCall) {
             String url = CATALOG_RESOURCE_URL.get();
             if (url.endsWith("/")) {
-                url = url.substring(0, url.length()-1);
-            }
-            Response resp = Request.Post(String.format("%s?refresh&action=refresh", url)).execute();
-            int status = resp.returnResponse().getStatusLine().getStatusCode();
-            if (status >= 400) {
-                throw new IOException("Failed to reload got [" + status + "]");
+                url = url.substring(0, url.length() - 1);
             }
+            Request.Post(String.format("%s?refresh&action=refresh", url)).execute().handleResponse((resp) -> {
+                int status = resp.getStatusLine().getStatusCode();
+                if (status >= 400) {
+                    log.warn("Skipping catalog refresh because it failed. Response code: {}, reason: {}", status, resp.getStatusLine().getReasonPhrase());
+                }
+                return null;
+            });
             firstCall = false;
```

Comment:
Yes, its acceptable/intentional and this doesnt change how it behaves because previous versions of catalog-service basically never returned anything other than 200 to refresh calls. The idea is is that we'll make one best effort attempt at updating and then move on. Up until now, catalog-service always returned 200 on refresh calls, even if the refresh failed, so this doesn't really change the functionality other than explicitly logging a warning.